compiler/src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -404,7 +406,7 @@ trait TypeAssigner {
         }
         else {
           val argTypes = preCheckKinds(args, pt.paramInfos).tpes
-          if (sameLength(argTypes, paramNames) || ctx.phase.prev.relaxedTyping) pt.instantiate(argTypes)
```

Comment:
I still think it'd be nice to keep the relaxedTyping check when we can (so, at least in the Typer).

Reply:
OK, I'll roll that into the next PR (#3366).